pkg/cvo/sync_worker.go

```diff
@@ -379,20 +386,26 @@ func (w *SyncWorker) Update(ctx context.Context, generation int64, desired confi
 		Overrides:  config.Spec.Overrides,
 	}
 
+	var priorCaps map[configv1.ClusterVersionCapability]struct{}
+
 	// the sync worker’s generation should always be latest with every change
 	if w.work != nil {
 		w.work.Generation = generation
+		priorCaps = w.work.Capabilities.EnabledCapabilities
 	}
 
 	if work.Empty() {
 		klog.V(2).Info("Update work has no release image; ignoring requested change")
 		return w.status.DeepCopy()
 	}
 
-	work.Capabilities = capability.SetCapabilities(config)
-	versionEqual, overridesEqual := equalSyncWork(w.work, work, fmt.Sprintf("considering cluster version generation %d", generation))
+	work.Capabilities = capability.SetCapabilities(config, priorCaps)
+	w.status.CapabilitiesStatus.ImplicitlyEnabledCaps = work.Capabilities.ImplicitlyEnabledCapabilities
```

Comment:
don't we want this down under the `equalSyncWork` guard if/when we set `w.work`?